djangocms_versioning/admin.py

```diff
@@ -16,23 +24,49 @@ def save_model(self, request, obj, form, change):
         super().save_model(request, obj, form, change)
         if not change:
             # create a new version object and save it
-            extension = apps.get_app_config('djangocms_versioning').cms_extension
-            version_model = extension.content_to_version_models[obj.__class__]
-            version_model.objects.create(content=obj)
+            Version.objects.create(content=obj)
 
     def get_queryset(self, request):
         """Limit query to most recent content versions
         """
         queryset = super().get_queryset(request)
-        versioning_extension = apps.get_app_config(
-            'djangocms_versioning').cms_extension
-        version_model = versioning_extension.content_to_version_models[
-            queryset.model]
-        filter_name = '{}__in'.format(version_model.__name__.lower())
-        latest_versions = version_model.objects.distinct_groupers()
-        return queryset.filter(**{filter_name: latest_versions})
+        versioning_extension = apps.get_app_config('djangocms_versioning').cms_extension
+        versionable = versioning_extension.versionables_by_content[queryset.model]
+        return queryset.filter(pk__in=versionable.distinct_groupers())
+
+
+class VersionChangeList(ChangeList):
```

Comment:
Correct me if I'm wrong, but I don't think this is tested at all. Either add an integration test and/or unit tests for each method. My gut feeling it that the best way to go is integration tests for the main functionality (i.e. adding grouper to the url query or not adding it and if we get the results we're expecting - I think the django client gives a way to access the queryset). And then unit tests for `IncorrectLookupParameters`.